File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int[] delete_list) {
        int[] answer = new int[arr.length];
        int index = 0;
        
        for(int i = 0; i < arr.length; i++){
            int arr_num = arr[i];
            boolean check = false;
            for(int j = 0; j < delete_list.length; j++){
                int delete_num = delete_list[j];
                if(arr_num == delete_num) {
                    check = true;
                    break;
                }
            }
            if(!check) answer[index++] = arr_num;
        }
        
        answer = Arrays.copyOfRange(answer, 0, index);
        
        return answer;
    }
}

File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len, int delete_list[], size_t delete_list_len) {
    
    int* answer = (int*)malloc(arr_len * sizeof(int));
    int delete_arr[1001] = {0};
    int idx = 0;

    for(int i = 0; i < delete_list_len; i++){
        delete_arr[delete_list[i]]++;
    }
    
    for(int j = 0; j < arr_len; j++){
        if(delete_arr[arr[j]] == 0){
            answer[idx++] = arr[j];
        }
    }
    
    return answer;
}